assets/javascripts/modules/checkout/voucherFields.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import moment from 'moment'
+import formElements from './formElements'
+import CustomDatePicker from '../react/customDatePicker'
+import CharacterCountedTextArea from './characterCountedTextArea'
+import reviewDetails from './reviewDetails'
+
+require('react-datepicker/dist/react-datepicker.css');
+
+const MAX_WEEKS_AVAILABLE = 4;
+
+function filterDate() {
+    return (date) => date.day() === 1;
+}
+
+function getFirstSelectableDate() {
+    var now = moment.utc();
+    var currentWeekday = now.weekday();
+    var currentHour = now.hour();
+    var mostRecentMonday = moment().startOf('isoWeek');
+    var weeksToAdd = currentWeekday >= 3 && currentHour >= 6 ? 3 : 2;
```

Comment:
It would also be good to break these numbers out at least into consts to explain what this check does